aiida_pseudo/data/pseudo/jthxml.py

```diff
@@ -30,14 +31,24 @@ def parse_element(stream: BinaryIO):
 class JthXmlData(PseudoPotentialData):
     """Data plugin to represent a pseudo potential in JTH XML format."""
 
-    def set_file(self, stream: BinaryIO, filename: str = None, **kwargs):  # pylint: disable=arguments-differ
-        """Set the file content.
+    def set_file(self, source: typing.Union[str, pathlib.Path, typing.BinaryIO], filename: str = None, **kwargs):  # pylint: disable=arguments-differ
+        """Set the file content and parse other optional attributes from the content.
 
-        :param stream: a filelike object with the binary content of the file.
+        .. note:: this method will first analyse the type of the ``source`` and if it is a filepath will convert it
+            to a binary stream of the content located at that filepath. This is then passed on to the superclass. This
+            needs to be done first, because it will properly set the file and filename attributes that are expected to
+            be there by other methods. Straight after the superclass call, the source seeker needs to be reset to zero
+            if it needs to be read again, because the superclass most likely will have read the stream to the end.
+            Finally it is important that the ``prepare_source`` is called here before the superclass invocation, because
+            this way the conversion from filepath to byte stream will be performed only once. Otherwise, each subclass
+            would perform the conversion over and over again.
+
+        :param source: the source pseudopotential content, either a binary stream, or a `str` or `Path` to the absolute
+            path of the file on disk.
         :param filename: optional explicit filename to give to the file stored in the repository.
         :raises ValueError: if the element symbol is invalid.
```

Comment:
This method also raises a `TypeError` in case the `source` type is not valid. Should we "bubble up" this `raises` information to this docstring from the `prepare_source` docstring?

Reply:
done